x-conduit/src/X/Control/Monad/Trans/Resource.hs

```diff
@@ -0,0 +1,23 @@
+{-# OPTIONS_GHC -fno-warn-orphans #-}
+module X.Control.Monad.Trans.Resource where
+
+import Control.Monad.Trans.Class (lift)
+import Control.Monad.Trans.Resource (MonadResource, liftResourceT)
+import Control.Monad.Trans.Either (EitherT)
+
+-- | This instance is useful when adding errors to the monad used for some conduit processing
+-- when resources, like files, are involved. For example:
+--
+-- foo :: MonadResource m => ConduitM () Void (EitherT Text m) [ByteString]
+-- foo = CB.sourceFile "src/Main.hs"
+--  =$= CB.lines
+--  -- filter some lines
+--  =$= CL.map parse
+--  -- collect errors in EitherT Text m
+--  =$= CL.mapM hoistEither
+--  =$= CL.consume
+--
+-- parse :: ByteString -> Either Text ByteString
+-- parse _ = Left ":("
+instance MonadResource m => MonadResource (EitherT e m) where
```

Comment:
Yeh, I was going to come back to this comment. If we want to switch to ExceptT I don't think there are any remaining roadblocks. The reason for EitherT in the start was packages that were dependent on transformers 3 which having a quick scan, looks like a non-issue now. I must say that I dislike the naming of ExceptT, but that isn't unusual (in terms of me not liking a name in haskell :/), and as a library it is far better supported across all library sub-ecosystems, where either avoids a lot of the pipes/conduit deps which is fair enough but does occasionally cause friction as per this change and one a while ago around MFunctor so it might be worth making the jump.